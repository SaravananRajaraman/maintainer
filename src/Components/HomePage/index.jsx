import React from 'react';
import {Grid,Button} from '@material-ui/core';
import RepoCard,{RepoSkeleton} from '../../Components/RepoCard';
import { connect } from "react-redux";
import styled from 'styled-components';
import TableChartIcon from '@material-ui/icons/TableChart';


const StyledBox = styled('div')({
  padding : '20px'
});

const StyledButton = styled(Button)`
      &&{
        margin-bottom : 20px;
        float : right;
      }
`

function HomePage(props){
      const {Repos,loading} = props;     
    return (
      <StyledBox>
          <StyledButton href='#/Table' startIcon={<TableChartIcon />}>View Table</StyledButton>         
          <Grid container spacing={2}>
              {loading ?
                  [0,1].map(value =>                         
                        <Grid key={value} item xs={12} sm={6} md={4}>
                            <RepoSkeleton/>
                  </Grid> ) :                    
                  Repos.map(value =>                         
                    <Grid key={value.name} item xs={12} sm={6} md={4}>
                        <RepoCard Repo={value}/>
                    </Grid>)}
          </Grid>
      </StyledBox>
    )
};

const mapStatesToProps = state => {
     return {
        Repos : state.home.Repos,
        loading : state.home.loading        
     } 
};

export default connect(mapStatesToProps)(HomePage);