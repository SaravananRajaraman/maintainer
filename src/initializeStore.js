import { createStore,applyMiddleware } from "redux";
import thunk from 'redux-thunk';
import createReducer from "./reducers";

let middleware = [ thunk ];

const initializeStore = () => {
  
  const store = createStore(    
    createReducer(),       
    applyMiddleware(...middleware)     
  );   
  return store;
};

export default initializeStore;