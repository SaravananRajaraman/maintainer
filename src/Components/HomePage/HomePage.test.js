import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import HomePage from './index';
import {findTestAttr} from '../../Utils';
import initializeStore from '../../initializeStore';

const store = initializeStore();

const setUp = (props={}) => {         
    const component = shallow(<HomePage store={store} {...props} />);    
    return component.dive().dive();
}

describe('Homepage Component', ()=>{    

    it('Should render the component without error',()=>{
        let component = setUp();        
        let wrapper = findTestAttr(component,'HomePage-Component');
        expect(wrapper.length).toBe(1); 
    });

    it('Should match the loading skeleton',()=>{
        let component = setUp();               
        expect(toJson(component)).toMatchSnapshot(); 
    });

    it('Should match the RepoCard',()=>{
        let Repos = [{name : 'name1', owner : [{id : 1, name : 'owner1', email : 'mail.com'},
                                               {id : 2, name : 'owner2', email : 'mail.com'}]},
                    {name : 'name2', owner : [{id : 3, name : 'owner3', email : 'mail.com'},
                                              {id : 4, name : 'owner4', email : 'mail.com'}]}];         
        let component = setUp();   
        component.setProps({loading : false, Repos : Repos});                
        expect(toJson(component)).toMatchSnapshot(); 
    });

});