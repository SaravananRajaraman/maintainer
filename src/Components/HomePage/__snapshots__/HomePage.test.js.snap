// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage Component Should match the RepoCard 1`] = `
<styled.div
  data-test="HomePage-Component"
>
  <Styled(WithStyles(ForwardRef(Button)))
    href="#/Table"
    startIcon={<UNDEFINED />}
  >
    View Table
  </Styled(WithStyles(ForwardRef(Button)))>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={2}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      key="name1"
      md={4}
      sm={6}
      xs={12}
    >
      <RepoCard
        Repo={
          Object {
            "name": "name1",
            "owner": Array [
              Object {
                "email": "mail.com",
                "id": 1,
                "name": "owner1",
              },
              Object {
                "email": "mail.com",
                "id": 2,
                "name": "owner2",
              },
            ],
          }
        }
        test-data="Repocard"
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      key="name2"
      md={4}
      sm={6}
      xs={12}
    >
      <RepoCard
        Repo={
          Object {
            "name": "name2",
            "owner": Array [
              Object {
                "email": "mail.com",
                "id": 3,
                "name": "owner3",
              },
              Object {
                "email": "mail.com",
                "id": 4,
                "name": "owner4",
              },
            ],
          }
        }
        test-data="Repocard"
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</styled.div>
`;

exports[`Homepage Component Should match the loading skeleton 1`] = `
<styled.div
  data-test="HomePage-Component"
>
  <Styled(WithStyles(ForwardRef(Button)))
    href="#/Table"
    startIcon={<UNDEFINED />}
  >
    View Table
  </Styled(WithStyles(ForwardRef(Button)))>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={2}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      key="0"
      md={4}
      sm={6}
      xs={12}
    >
      <RepoSkeleton />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      key="1"
      md={4}
      sm={6}
      xs={12}
    >
      <RepoSkeleton />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</styled.div>
`;
